code/datums/brain_damage/creepy_trauma.dm

```diff
@@ -0,0 +1,117 @@
+/datum/brain_trauma/special/creep
+	name = "Erotomania"
+	desc = "Patient has a subtype of delusional disorder, becoming irrationally attached to someone."
+	scan_desc = "severe erotomaniac delusions"
+	gain_text = ""
+	lose_text = "<span class='warning'>You no longer feel so attached.</span>"
+	can_gain = TRUE
+	resilience = TRAUMA_RESILIENCE_SURGERY
+	var/mob/living/obsession
+	var/datum/objective/spendtime/attachedcreepobj
+	var/datum/antagonist/creep/antagonist
+
+	var/total_time_creeping = 0 //just for roundend fun
+	var/time_spent_away = 0
+	var/obsession_hug_count = 0
+
+/datum/brain_trauma/special/creep/on_gain()
+
+	//setup, linking, etc//
+	..()
+	if(!obsession)//admins didn't set one
+		obsession = find_obsession()
+		if(!obsession)//we didn't find one
+			qdel(src)
+	owner.mind.add_antag_datum(/datum/antagonist/creep)
+	antagonist = owner.mind.has_antag_datum(/datum/antagonist/creep)
+	antagonist.trauma = src
+	//antag stuff, //
+	antagonist.forge_objectives(obsession.mind)
+	antagonist.greet()
+
+
+
+/datum/brain_trauma/special/creep/on_life()
+	if(!obsession)//gibbing, etc
+		return
+	if(obsession.stat == DEAD) //killing them "cures" you! kinda! ish!
+		return
+	var/foundyou = FALSE
+	for(var/mob/living/L in range(7, owner))
+		if(L == obsession)
+			foundyou = TRUE
+			break
+	if(foundyou)
+		SEND_SIGNAL(owner, COMSIG_ADD_MOOD_EVENT, "creeping", /datum/mood_event/creeping, obsession.name)
+		total_time_creeping += 20
+		time_spent_away = 0
+		if(attachedcreepobj)
+			attachedcreepobj.timer -= 20 //mob subsystem ticks every 2 seconds(?), remove 20 deciseconds from the timer. sure, that makes sense.
```

Comment:
what is the purpose of this

Reply:
if they have the objective to be near the target while they are alive, it will set the trauma's attachedcreepobj to itself. I was already counting when they were near and what mood event to give if they were away long enough, it was easy to add in this as well

I should comment that though, it isn't clear